api/src/main/java/com/epam/pipeline/manager/git/github/GitHubServerApi.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2024 EPAM Systems, Inc. (https://www.epam.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.manager.git.github;
+
+import com.epam.pipeline.entity.git.github.GitHubCommitNode;
+import com.epam.pipeline.entity.git.github.GitHubContent;
+import com.epam.pipeline.entity.git.github.GitHubRef;
+import com.epam.pipeline.entity.git.github.GitHubRelease;
+import com.epam.pipeline.entity.git.github.GitHubRepository;
+import com.epam.pipeline.entity.git.github.GitHubSource;
+import com.epam.pipeline.entity.git.github.GitHubTag;
+import com.epam.pipeline.entity.git.github.GitHubTagRequest;
+import com.epam.pipeline.entity.git.github.GitHubTree;
+import retrofit2.Call;
+import retrofit2.http.Body;
+import retrofit2.http.DELETE;
+import retrofit2.http.GET;
+import retrofit2.http.HTTP;
+import retrofit2.http.PATCH;
+import retrofit2.http.POST;
+import retrofit2.http.PUT;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+import java.util.List;
+
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public interface GitHubServerApi {
```

Comment:
Let's rename class to `GitHubApi` (usually we name such classes like `<some service>Api`)